src/auth/identitypoolclient.ts

```diff
@@ -97,53 +98,67 @@ export class IdentityPoolClient extends BaseExternalAccountClient {
 
     const opts = originalOrCamelOptions(options as IdentityPoolClientOptions);
     const credentialSource = opts.get('credential_source');
-    const credentialSourceOpts = originalOrCamelOptions(credentialSource);
-
-    const formatOpts = originalOrCamelOptions(
-      credentialSourceOpts.get('format')
-    );
-
-    // Text is the default format type.
-    const formatType = formatOpts.get('type') || 'text';
-    const formatSubjectTokenFieldName = formatOpts.get(
-      'subject_token_field_name'
-    );
-
-    if (formatType !== 'json' && formatType !== 'text') {
-      throw new Error(`Invalid credential_source format "${formatType}"`);
-    }
-    if (formatType === 'json' && !formatSubjectTokenFieldName) {
+    const subjectTokenSupplier = opts.get('subject_token_supplier');
+    if (!credentialSource && !subjectTokenSupplier) {
```

Comment:
nit: reduce nesting here and below as well

[go/tott/651](https://goto.google.com/tott/651)

Reply:
not sure what nesting you are referring to here? There isn't anything deeper than 2 levels. If you mean having the errors here be in if statements instead of if else like you suggested in the AWS client, made that change.

Reply:
Yep, thats the change I was suggesting. Thanks.